fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -182,3 +219,45 @@ def load_data(self, X, y, **kwargs):
         super().load_data(X, y,
                           event=pd.Series(y).apply(lambda y: _LABEL + "=" + str(y)),
                           **kwargs)
+
+
+class ErrorRateRatio(ConditionalSelectionRate):
+    r"""Implementation of Error Rate Ratio as a moment.
+
+    Measures the ratio in errors per attribute by overall error.
+    The 2-sided version of error ratio can be written as
+    ratio <= error(A=a) / total_error <= 1/ratio
+    .. math::
+    ratio <= E[abs(h(x) - y) = 1 | A = a] / E[abs(h(x) - y) = 1] <= 1/ratio\; \forall a
+
+    This implementation of :class:`ConditionalSelectionRate` defines
+    events corresponding to the unique values of the `Y` array.
+
+    The `prob_event` :class:`pandas:pandas.DataFrame` will record the
+    fraction of the samples corresponding to each unique value in
+    the `Y` array.
+
+    The `index` MultiIndex will have a number of entries equal to
+    the number of unique values for the sensitive feature, multiplied by
+    the number of unique values of the `Y` array, multiplied by two (for
+    the Lagrange multipliers for positive and negative constraints).
+
+    With these definitions, the :math:`signed_weights` method
+    will calculate the costs according to Example 4 of
+    `Agarwal et al. (2018) <https://arxiv.org/abs/1803.02453>`_.
+
+    The `ratio` defines the amount of relaxation that is allowed for the
+    constraint. The value varies between 0 and 1. The ratio of 1 means,
+    the constraint is given no relaxation and thus, the constraint tries to
+    evaluate for
+    error(A=a) / total_error = 1
+    """
+
+    short_name = "ErrorRateRatio"
+
+    def load_data(self, X, y, **kwargs):
+        """Load the specified data into the object."""
+        super().load_data(X, y,
+                          event=_ALL,
+                          utility=pd.Series({_POSITIVE_UTILITY: (1-y), _NEGATIVE_UTILITY: y}),
```

Comment:
Series is a one-dimensional datastructure, so perhaps DataFrame is more appropriate?

Reply:
same earlier

Reply:
I'll change it to a DataFrame.